analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/InvocationCheck.cs

```diff
@@ -26,27 +29,49 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn.Checks
 {
     internal class InvocationCheck : SymbolicCheck
     {
+        private static ImmutableArray<KnownType> throwHelperAttributes = new[]
+        {
+            KnownType.System_Diagnostics_CodeAnalysis_DoesNotReturnAttribute,
+            KnownType.JetBrains_Annotations_TerminatesProgramAttribute,
```

Comment:
There is an inconsistency here:
For "ValidatedNotNull" we only check for the class name but not the namespace:

`attribute.AttributeClass?.Name == "ValidatedNotNullAttribute"`

Here we also test for the namespace. What do you think is the better way forward for attribute support:
* [ ] always by name
* [ ] always by fully qualified name
* [ ] case-by-case

Reply:
Here the attributes are framework specific and not the same (by coincidence). So well-known full name is the best way